src/nvim/api/keysets.h

```diff
@@ -32,6 +32,7 @@ typedef struct {
   Boolean virt_text_hide;
   Boolean hl_eol;
   String hl_mode;
+  Boolean invalidate;
```

Comment:
I think this should be a flag `MT_FLAG_INVALIDATE` on MTKey itself, just like `MT_FLAG_RIGHT_GRAVITY`. This is really a property of the extmark itself and useful even without visible decorations, i e plugin can check that the text containing the mark was deleted.

Reply:
Yeah makes sense, was wondering about this myself as well.